lightning/src/routing/router.rs

```diff
@@ -4409,9 +4413,9 @@ mod tests {
 		'load_endpoints: for _ in 0..10 {
 			loop {
 				seed = seed.overflowing_mul(0xdeadbeef).0;
-				let src = nodes.keys().skip(seed % nodes.len()).next().unwrap();
+				let src = &PublicKey::from_slice(nodes.keys().skip(seed % nodes.len()).next().unwrap().as_slice()).unwrap();
```

Comment:
How about `impl From<NodeId> for PublicKey` (and vice-versa)?

Reply:
I don't believe we can implement `From` for a struct outside our crate. There's a workaround with using a tuple struct wrapping the `PublicKey`. But also note that the conversion functions consume the struct being converted, so in many cases we'd have to make a copy since we only have a reference, which I think we'd want to avoid.

Reply:
@jkczyz since `NodeId` is in this crate `From<T> for NodeId` can always be implemented for any `T` we want. Also since Rust 1.41 `From<NodeId> for T` is possible. In older versions at least `impl Into<T> for NodeId`.

Also `From` and `Into` can be implemented for references, so copying should be non-issue.

Reply:
I think we can leave it as is for now then?

Reply:
I don't care much, just note that Debian **oldstable** contains Rust 1.41, which is sufficient to solve the `From` issue. Debian Bullseye has 1.48 which can even support modern Tokio. Debian stable is probably the most conservative distro that's actually widely used, so doesn't seem too bad to me.